spec/github_helper_spec.rb

```diff
@@ -35,63 +35,6 @@
     end
   end
 
-  describe 'github_token' do
-    after do
-      ENV['GHHELPER_ACCESS'] = nil
-      ENV['GITHUB_TOKEN'] = nil
-    end
-
-    it 'can use `GHHELPER_ACCESS`' do
-      ENV['GHHELPER_ACCESS'] = 'GHHELPER_ACCESS'
-      expect(described_class.github_token!).to eq('GHHELPER_ACCESS')
-    end
-
-    it 'can use `GITHUB_TOKEN`' do
-      ENV['GITHUB_TOKEN'] = 'GITHUB_TOKEN'
-      expect(described_class.github_token!).to eq('GITHUB_TOKEN')
-    end
-
-    it 'prioritizes GHHELPER_ACCESS` over `GITHUB_TOKEN` if both are present' do
-      ENV['GITHUB_TOKEN'] = 'GITHUB_TOKEN'
-      ENV['GHHELPER_ACCESS'] = 'GHHELPER_ACCESS'
-      expect(described_class.github_token!).to eq('GHHELPER_ACCESS')
-    end
-
-    it 'prints an error if no environment variable is present' do
-      expect { described_class.github_token! }.to raise_error(FastlaneCore::Interface::FastlaneError)
-    end
-  end
-
-  describe 'github_client' do
-    let(:client) do
-      instance_double(
-        Octokit::Client,
-        user: instance_double('User', name: 'test'),
-        'auto_paginate=': nil
-      )
-    end
-
-    before do
-      allow(described_class).to receive(:github_token!).and_return('')
-      allow(Octokit::Client).to receive(:new).and_return(client)
-    end
-
-    after do
-      # Clean up the client memoization between runs to ensure it's re-initialized in each test
-      described_class.remove_class_variable(:@@client) if described_class.class_variable_defined?(:@@client)
-    end
-
-    it 'is not nil' do
-      expect(described_class.github_client).not_to be_nil
-    end
-
-    it 'memoizes the client' do
-      expect(Octokit::Client).to receive(:new).once
-      described_class.github_client
-      described_class.github_client
-    end
-  end
-
```

Comment:
Self-Review: As we removed this from the class **GithubHelper**, those tests were not needed anymore.

Reply:
While that's true, it would be nice, as a counter part, to add tests to cover:
 - The fact that the `@client` in the `GithubHelper` gets passed the github token that was passed in `initialize` (a bit trivial of a thing to test, but 🤷 )
 - Ensure that the fallback mechanism works on the actions.
    - In particular, ideally we'd have specs on each of the actions that were touched by this PR and on which a `ConfigItem` was added for that token, mock the client, and ensure that when we call the action with different values of for the `access_token:` option, the `GithubHelper` get the expected token (i.e. with fallback and all) as parameter to `new`. This is a bit debatable because in a way that'd be testing `ConfigItem` itself (which is already covered by fastlane's own tests), but on the other hand this would also ensure that we do use those `ConfigItem` values (and don't ignore it in the action's implementation) and pass the token properly down to the helper.
    - Sadly, many of our current actions don't have unit tests as of now, especially the actions we created ages ago (when we were not as fluent in Ruby nor as rigorous), so they might not all have existing specs for you to add those additional tests to… but otoh that could be a great occasion to kick off minimal specs for them. Gotta start somewhere!

Reply:
>The fact that the @client in the GithubHelper gets passed the github token that was passed in initialize (a bit trivial of a thing to test, but 🤷 )

Yeah, a trivial test just to know if the OktoClient is receiving the token correctly, I see value in it. We never know what we will be updating 2 years from now 😄 

>In particular, ideally we'd have specs on each of the actions that were touched by this PR and on which a ConfigItem was added for that token, mock the client, and ensure that when we call the action with different values of for the access_token: option, the GithubHelper get the expected token (i.e. with fallback and all) as parameter to new. This is a bit debatable because in a way that'd be testing ConfigItem itself (which is already covered by fastlane's own tests), but on the other hand this would also ensure that we do use those ConfigItem values (and don't ignore it in the action's implementation) and pass the token properly down to the helper.

Yeah, I tend to agree with that rationale that Testing the ConfigItem is kinda retesting what Fastlane already do, but I find it valuable to have at least the config of our ConfigItem covered, so we can assure that this shared ConfigItem is on the way we want it to be.

>Sadly, many of our current actions don't have unit tests as of now, especially the actions we created ages ago (when we were not as fluent in Ruby nor as rigorous), so they might not all have existing specs for you to add those additional tests to… but otoh that could be a great occasion to kick off minimal specs for them. Gotta start somewhere!

No worries I'll write some unit tests for the actions that I touch, At least to have the parameter and the ConfigItem presence assured. About how to test the fallback or the error I'm not sure yet how to test it or expect this error. I'll use the weekend and Monday to research about it 😄